### PR TITLE
Add Effort section to DG

### PR DESCRIPTION
**Pull request by charliemoweng**

### CODE REVIEW

**Comment by felissaf** [comment]
Wow this is like a legit essay so good🥲👍🏻👍🏻👍🏻 Thanks for doing this!!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -892,3 +892,33 @@ testers are expected to do more *exploratory* testing.
...
+Other challenges we had included dealing with testing and storage. For many of us, this was the first time handling these aspects of coding.
+It was not easy, especially for those members who volunteered as trailblazers and set the standard for writing test cases and storage-related logic. 
+Kudos to them!
+
+Overall, this project posed many difficulties and challenges that we had to overcome.
+To many of us, this was the first collaborative software engineering project that involved facets that were once unfamiliar, such as testing, UI and storage.
+It would be an understatement to say that it was not easy.
+Nevertheless, we, together as a team, rose above all these obstacles and passed this trial of diligence, determination, resourcefulness, and teamwork. 
+
+Thank you for reading until here. Have a nice day! : )
```
Comment (WangGLJoseph) [inline docs/DeveloperGuide.md:924]: Not sure if this line is proper for the DG haha

**Review by WangGLJoseph — COMMENTED** [review]
Looking good!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (charliemoweng) [reply docs/DeveloperGuide.md:924]: Haha adding a bit of personal flair. i can remove it if not appropriate.

**Comment by Ruofan1023** [comment]
Thanks man! LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (charliemoweng) [reply docs/DeveloperGuide.md]: Removed

**Pull request merged**